packages/flutter/lib/src/material/app_bar.dart

```diff
@@ -1154,6 +1155,17 @@ class SliverAppBar extends StatefulWidget {
   /// Defaults to [NavigationToolbar.kMiddleSpacing].
   final double titleSpacing;
 
+
+  /// The size of the app bar when it is collapsed.
+  ///
+  /// By default, the collapsed height is set to [kToolbarHeight]. If the
```

Comment:
It's not clear if "the size of the app bar when it is collapsed" is widget.collapsedHeight + [bottom.preferredSize.height] etc, or if it's just widget.collapsedHeight. It looks like you meant the latter, but maybe it should be the former?

Reply:
Makes sense.

## Collapsed height
After looking more at it, I think collapsedHeight should set the collapsed height excluding any padding. It is similar to how the expandedHeight works:

![Sliver app bar](https://user-images.githubusercontent.com/1770678/83883744-2fd00c80-a744-11ea-948c-b4e78390abee.png)

The user will have to make sure that they do not set a too small of a collapsedHeight, as that could cause overflow error for the bottom widget or hide the title. I am not sure if we need to add any assertions for this, I do not think it should be needed as it is quite obvious.

## Opacity change
I noticed also an issue with how the opacity changes when a user scrolls up, it does not correctly change the opacity for the title in my current implementation, so I will fix this too:
![no_opacity_change](https://user-images.githubusercontent.com/1770678/83883938-7887c580-a744-11ea-9218-8024364f5642.gif)

Here is how it looks like without collapsed height and the opacity changes correctly:
![without_collapsed_height](https://user-images.githubusercontent.com/1770678/83884386-167b9000-a745-11ea-8b2d-58cec2235661.gif)